Flaky test: publication-time.test.js fails due to timing race condition

## Description

The `tests/blocks/publication-time/publication-time.test.js` test is failing intermittently due to a timing race condition.

## Failure Pattern

```
AssertionError: expected '2025-10-29T11:08:06.000Z' to equal '2025-10-29T11:08:07.000Z'
+ expected - actual

-2025-10-29T11:08:06.000Z
+2025-10-29T11:08:07.000Z

at n.<anonymous> (tests/blocks/publication-time/publication-time.test.js:18:44)
```

## Root Cause

The test is comparing timestamps with exact equality, but depending on when the test executes (before vs. after a second boundary), the timestamps can differ by 1 second. This is a classic flaky test pattern.

## Impact

- Unrelated PRs fail CI randomly
- Example: PR #963 failed with this test despite not touching publication-time code
- False negatives block legitimate changes

## Evidence

- Failed in PR #963: https://github.com/adobe/helix-website/pull/963
- Build log: https://github.com/adobe/helix-website/actions/runs/18905791823/job/53963679224

## Suggested Fix

The test at line 18 should allow for a small time window (e.g., ±1-2 seconds) instead of exact equality, or use a mocked/fixed time for deterministic testing.

## Labels

This is a flaky test that should be fixed or temporarily skipped to prevent blocking unrelated PRs.

--- a/tests/blocks/publication-time/publication-time.test.js
+++ b/tests/blocks/publication-time/publication-time.test.js
@@ -11,11 +11,18 @@ document.body.innerHTML = mock;
 describe('Publication time block', async () => {
   it('has a relative-time element', async () => {
     const block = document.querySelector('.block');
-    decorate(block);
-    const rt = document.querySelector('relative-time');
+    // Capture lastModified BEFORE calling decorate to avoid timing race condition
     const lastMod = new Date(document.lastModified);
     const shortDate = lastMod.toLocaleDateString();
-    expect(rt.getAttribute('datetime')).to.equal(lastMod.toISOString());
+    decorate(block);
+    const rt = document.querySelector('relative-time');
+    const rtDatetime = rt.getAttribute('datetime');
+
+    // Allow for small time differences (±2 seconds) to handle timing edge cases
+    const rtDate = new Date(rtDatetime);
+    const timeDiff = Math.abs(rtDate.getTime() - lastMod.getTime());
+    expect(timeDiff).to.be.at.most(2000, `Time difference was ${timeDiff}ms`);
+
     expect(rt.textContent).to.equal(shortDate);
     expect(block.textContent).to.equal(`prefix for relative time ${shortDate}`);
   });

